examples/silentpayments.c

```diff
@@ -0,0 +1,432 @@
+        }
+
+        {
+            /*** Scanning with labels as a full node (Bob) ***
+             *
+             * Since Bob has access to the full transaction, scanning is simple:
+             *
+             *     1. Collect the relevant public data from the transaction inputs and call
+             *        `secp256k1_silentpayments_recipient_public_data_create`
+             *     2. Call `secp256k1_silentpayments_recipient_scan_outputs`
+             *
+             * Additionally, since Bob has access to the full transaction outputs when scanning its easy for him
+             * to scan with labels, as demonstrated below. For efficient scanning, Bob keeps a cache of
+             * every label he has previously used and uses a callback to check if a potential label exists
+             * in his cache. Since the labels are created using an incremental integer `m`, if Bob ever
+             * forgets how many labels he has previously used, he can pregenerate a large number of
+             * labels e.g., 0..100_000.
+             */
+            secp256k1_silentpayments_found_output found_outputs[N_TX_OUTPUTS];
+            secp256k1_silentpayments_found_output *found_output_ptrs[N_TX_OUTPUTS];
+            secp256k1_silentpayments_public_data public_data;
+            secp256k1_pubkey spend_pubkey;
+            size_t n_found_outputs;
+            unsigned int m = 1;
+            struct labels_cache labels_cache;
+
+            for (i = 0; i < N_TX_OUTPUTS; i++) {
+                found_output_ptrs[i] = &found_outputs[i];
+            }
+
+            /* In this contrived example, our label context needs the secp256k1 context because our lookup function
+             * is calling `secp256k1_ec_pubkey_cmp`. In practice, this context can be anything the lookup function needs.
+             */
+            labels_cache.ctx = ctx;
+
+            /* Load Bob's spend public key */
+            ret = secp256k1_ec_pubkey_parse(ctx, &spend_pubkey, bob_spend_pubkey, 33);
+
+            /* Add an entry to the cache. This implies Bob has previously called `secp256k1_silentpayments_recipient_create_labelled_spend_pubkey`
+             * and is using the resulting labelled spend pubkey to encode a labelled silent payments address.
+             */
+            ret = secp256k1_silentpayments_recipient_create_label_tweak(ctx,
+                &labels_cache.entries[0].label,
+                labels_cache.entries[0].label_tweak,
+                bob_scan_key,
+                m
+            );
+            assert(ret);
+            labels_cache.entries_used = 1;
+
+            /* Bob collects the public data from the transaction inputs and creates a `secp256k1_silentpayments_public_data` object.
+             * He uses this for his own scanning and also serializes the `public_data` object to send to light clients. We will
+             * use this later for Carol, who is scanning as a light client. Note, anyone can create and vend these `public_data`
```

Comment:
nit: "vend" is a word that I had to look up, which is a bit uncommon for me, even as a non-native speaker. Maybe you can find a more common word. Perhaps just "provide"?

Reply:
Fixed.